test/value_from.cpp

```diff
@@ -183,6 +186,13 @@ class value_from_test
             BOOST_TEST(c.is_array());
             BOOST_TEST(to_string(c) == to_string(b));
         }
+        {
+            key_value_pair a{"2", 1};
+            value b{"2", 1};
+            value c = value_from(a);
+            BOOST_TEST(c.is_array());
+            BOOST_TEST(to_string(c) == to_string(b));
+        }
```

Comment:
@sdkrystian pointed out to me that `value_from` has special support for tuple-like types, hence I added this
However I am not a 100% sure whether `value_from(key_value_pair)` should yield an object with a single field, or an array with two elements (the current default).

Reply:
Two element array is correct.